desktop/ui/src/main/java/org/datacleaner/windows/DataHubDatastoreDialog.java

```diff
@@ -264,6 +286,11 @@ protected boolean validateForm() {
             return false;
         }
 
+        if (_tenantName == null) {
+            setStatusError("Please test the connection first");
+            return false;
```

Comment:
That is true, but the alternatives looked worse. We tried getting the alias when you hit the Save button. But when you then fail to get the tenant name (e.g. because the credentials are wrong), then you are faced with an exception in the Save button handling.
Retrieving the tenant name requires most of the other settings to be already correct. So in the end the best option seemed to be to always require testing the connection before saving.